### DISCUSSION TITLE
gnmic/kafka

### DISCUSSION THREAD

**Discussion by pperesbr** [karimra/gnmic#323] [General]
Hello,

Please, can you provide a full example with Kafka input/output?

Tks,

Paulo

**Comment by karimra** [comment]
Hi,

The kafka input/output docs pages should have all you need: 
https://gnmic.kmrd.dev/advanced/inputs/kafka_input/
https://gnmic.kmrd.dev/advanced/multi_outputs/kafka_output/

If you could give more details about what you want to do, I can come up with a more detailed config.

Karim

**Comment by pperesbr** [comment]
Hi,

I had already looked at the documentation you sent, thanks anyway.

Let me explain to you what I intend to do, please see if it is viable using
your project.

I'm setting up a base of equipment inventories using NodeJs / GraphQL
(backend), with the equipment registered in the database (ip, equipment
name, supplier) to search for the other equipment information (installed
cards, configured interfaces, etc.).

I'm studying gNMI and found your project on the internet, as it's already
working, I don't see why to redo, in general, what I thought about doing,
from my backend send a request via Kakfa to the GNMIC (input with ip,
credentials and which information retrieve (path)) and thus the GNMIC
(output) returns to the backend via Kafka.

Do you think it is possible?

Thanks

Paulo

On Fri, Feb 26, 2021 at 5:25 AM Karim Radhouani <notifications@github.com>
wrote:

> Hi,
>
> The kafka input/output docs pages should have all you need:
> https://gnmic.kmrd.dev/advanced/inputs/kafka_input/
> https://gnmic.kmrd.dev/advanced/multi_outputs/kafka_output/
>
> If you could give more details about what you want to do, I can come up
> with a more detailed config.
>
> Karim
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/karimra/gnmic/discussions/323#discussioncomment-408348>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTWWZK6TKPMKMVES3EIUK3TA5LHBANCNFSM4YH5BCQQ>
> .
>


-- 
Paulo Henrique Peres

**Reply by karimra** [reply]
Hi, 

kafka input is meant for actual gnmi messages not for loading gnmi targets.

gnmic will support a restful api in the next release, it will support a limited set of endpoints, amongst which are Add/Delete target config and start stop target subscriptions.

I believe you could add a target and start its subscription using rest calls, and read the output from kafka(or other)
I will keep you posted once it's released, and add a basic endpoints documentation.

Karim

**Comment by pperesbr** [comment]
Thanks Karim,

I will wait for the new version, I am still at the beginning of my backend
development.

Congratulations on the GNMIC development initiative.



On Mon, Mar 1, 2021 at 1:30 AM Karim Radhouani <notifications@github.com>
wrote:

> Hi,
>
> kafka input is meant for actual gnmi messages not for loading gnmi targets.
>
> gnmic will support a restful api in the next release, it will support a
> limited set of endpoints, amongst which are Add/Delete target config and
> start stop target subscriptions.
>
> I believe you could add a target and start its subscription using rest
> calls, and read the output from kafka(or other)
> I will keep you posted once it's released, and add a basic endpoints
> documentation.
>
> Karim
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/karimra/gnmic/discussions/323#discussioncomment-416611>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTWWZMGZ7V5LYAS6VNEPWDTBMJ7ZANCNFSM4YH5BCQQ>
> .
>


-- 
Paulo Henrique Peres

**Reply by karimra** [reply]
@dobicz Thanks, glad you like it. 

Can you check the v0.8.0 release?

- How to enable the API endpoint: https://gnmic.kmrd.dev/user_guide/api/api_intro/

- Add a target's configuration: https://gnmic.kmrd.dev/user_guide/api/configuration/#post-configtargets
The expected json body has the same format as what you would write in the config file under `targets`

- Start a target's subscriptions: https://gnmic.kmrd.dev/user_guide/api/targets/#post-targetsid

- Stop a target's subscriptions: https://gnmic.kmrd.dev/user_guide/api/targets/#delete-targetsid

- Delete a target's configuration: https://gnmic.kmrd.dev/user_guide/api/configuration/#delete-configtargetsid
This will also stop the target's subscriptions if there are active ones.
